Scripts/Singletons/SceneSwapper.gd

```diff
@@ -107,3 +115,38 @@ func change_to_level_scene(level_select, button_name):
 	var scene_key = stage_type + " " + level_number
 	
 	change_scene(scene_key)
+
+
+func init_scene_array():
+	scene_array.clear()
+	load_file_contents("res://Scenes/Levels/Tutorial")
+	load_file_contents("res://Scenes/Levels/Functions")
+	load_file_contents("res://Scenes/Levels/Recursion")
+	load_file_contents("res://Scenes/Levels/Control_Flow")
+	load_file_contents("res://Scenes/Levels/Data_Structures")
+
+
+func load_file_contents(path):
+	var dir = Directory.new()
+	
+	if dir.open(path) == OK:
+		dir.list_dir_begin()
+		var file_name = dir.get_next()
+		
+		while file_name != "":
+			if !dir.current_is_dir():
+				var scene_path = path + "/" + file_name
+				scene_array.push_back(scene_path)
+			
+			file_name = dir.get_next()
+	else:
+		print("An error occurred when trying to access the path.")
+
+
+func change_to_next_level_scene(old_scene):
+	var index = scene_array.find(old_scene, 0) + 1
+	
+	# Last level should not go back to first tutorial level (index = -1 + 1 if next scene isn't found)
```

Comment:
line 149 comment, wouldn't index = -1 only if old_scene can't be found? If it was the last level then it would be equal to scene_array.size(). The code checks out, but this comment might be confusing later on

Reply:
Comment out of date my bad